operator/api/v1alpha1/service_types.go

```diff
@@ -25,19 +25,32 @@ import (
 
 // ServiceSpec defines the desired state of Service
 type ServiceSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-
-	// Foo is an example field of Service. Edit service_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	ApplicationName    string    `json:"application,omitempty"`
+	PreDeplymentChecks EventSpec `json:"preDeploymentChecks"`
 }
 
 // ServiceStatus defines the observed state of Service
 type ServiceStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	Phase                   ServicePhase `json:"phase"`
+	PreDeploymentChecksName string       `json:"preDeploymentChecksName"`
 }
```

Comment:
I would also expect the state of the pre-deployment checks (pending, in progress, passed, failed) per pre-deployment check here.

Reply:
That's basically a phase, we have a 1on1 binding between Service CRD and Event CRD

Reply:
For the phase, it's ok. But I think one service will have multiple pre-deployment checks (or multiple experiments and one analysis) ...